### PR TITLE
Mouse movements should "take over" from keyboard focus in Combobox dropdown list, and vice versa, acting like a native select.

### LINKED ISSUE #2165: [BUG] To options kan få fokus om man benytter mus og tastatur på samme tid
## Hva er feilen/problemet?

Når jeg både bruker mus og tastatur for å velge option i ComboBox, så blir to options uthevet. 

- Åpne optionslisten
- Ta musen over et valg, litt ned på listen, uten å trykke på det. Nå skal dette være uthevet
- Trykk pil ned på tastaturet, nå skal første option også bli uthevet
- Se nå at to options er uthevet, både det første og det som var under der musepekeren sto
- Om man trykker flere ganger pil ned og deretter opp igjen, så vil fokus på begge options bevege seg opp/ned, og det blir litt vanskelig å følge med på hvem som egentlig har fokus.
- Om jeg flytter musepeker vekk fra combo box, så skjer ikke dette lengre

Når jeg tester native select, så håndterer denne bruk av både mus og tastatur samtidig, slik at det bare er et valg som får fokus.

**Med musepeker over et valg**

Skjermbilde 2023-08-15 kl  10 40 14

**Trykker pil ned, første option får fokus, samtidig som det som er under der musepeker sto også for fokus:**

Skjermbilde 2023-08-15 kl  10 40 28

### PR DESCRIPTION
**Pull request by it-vegard**
### Description

Move movements "take over" from keyboard focus in Combobox dropdown list, and vice versa, acting like a native select.

Fixes #2165

### Change summary

- When moving the mouse over a list item, the virtual keyboard focus is also moved to that item.
- If hovering over an item, but moving focus with the arrow keys, the hover styling disappears.

### CODE REVIEW

**Comment by it-vegard** [comment]
Should add some tests on Monday, as well as a changelog item.

File: @navikt/core/css/form/combobox.css [hunk 1]
```diff
@@ -256,15 +256,15 @@
 }
 
 .navds-combobox__list-item--focus,
-.navds-combobox__list-item:hover {
+.navds-combobox__list--with-hover .navds-combobox__list-item:hover {
   background-color: var(--ac-combobox-list-item-hover-bg, var(--a-surface-hover));
   cursor: pointer;
   border-left: 4px solid var(--ac-combobox-list-item-hover-border-left, var(--a-border-strong));
   padding-left: calc(var(--a-spacing-3) - 4px);
```
Comment (KenAJoh) [inline @navikt/core/css/form/combobox.css:263]:
```suggestion
  padding-left: calc(var(--a-spacing-3) - var(--spacing-4));
```

File: @navikt/core/css/form/combobox.css [hunk 2]
```diff
@@ -256,15 +256,15 @@
...
 }
 
 .navds-form-field--small .navds-combobox__list-item--focus,
-.navds-form-field--small .navds-combobox__list-item:hover {
+.navds-combobox__list--with-hover .navds-form-field--small .navds-combobox__list-item:hover {
   padding-left: calc(var(--a-spacing-2) - 4px);
```
Comment (KenAJoh) [inline @navikt/core/css/form/combobox.css:268]:
```suggestion
  padding-left: calc(var(--a-spacing-2) - var(--a-spacing-4));
```

File: @navikt/core/css/form/combobox.css [hunk 3]
```diff
@@ -277,7 +277,7 @@
 }
 
 .navds-combobox__list-item--selected.navds-combobox__list-item--focus,
-.navds-combobox__list-item--selected:hover {
+.navds-combobox__list--with-hover .navds-combobox__list-item--selected:hover {
   background-color: var(--ac-combobox-list-item-selected-hover-bg, var(--a-surface-action-subtle-hover));
   border-left: 4px solid var(--ac-combobox-list-item-selected-hover-border-left, var(--a-border-focus));
   padding-left: calc(var(--a-spacing-3) - 4px);
```
Comment (KenAJoh) [inline @navikt/core/css/form/combobox.css:283]:
```suggestion
  padding-left: calc(var(--a-spacing-3) - var(--a-spacing-4));
```

**Review by KenAJoh — APPROVED** [review]

**Pull request merged**